ci/github-script/reviewers.js

```diff
@@ -0,0 +1,129 @@
+// Caching the list of team members saves API requests when running the bot on the schedule and
+// processing many PRs at once.
+const members = {}
+
+async function handleReviewers({
+  github,
+  context,
+  core,
+  log,
+  dry,
+  pull_request,
+  reviews,
+  maintainers,
+  owners,
+  getTeamMembers,
+  getUser,
+}) {
+  const pull_number = pull_request.number
+
+  const users = new Set([
+    ...(await Promise.all(
+      maintainers.map(async (id) => (await getUser(id)).login),
+    )),
+    ...owners.filter((handle) => handle && !handle.includes('/')),
+  ])
+  log('reviewers - users', Array.from(users).join(', '))
+  console.log('users', Array.from(users))
+
+  const teams = new Set(
+    owners
+      .map((handle) => handle.split('/'))
+      .filter(([org, slug]) => org === context.repo.owner && slug)
+      .map(([, slug]) => slug),
+  )
+  log('reviewers - teams', Array.from(teams).join(', '))
+
+  const team_members = new Set(
+    (await Promise.all(Array.from(teams, getTeamMembers)))
+      .flat(1)
+      .map(({ login }) => login),
+  )
+  log('reviewers - team_members', Array.from(team_members).join(', '))
+
+  const new_reviewers = users
+    .union(team_members)
+    // We can't request a review from the author.
+    .difference(new Set([pull_request.user?.login]))
+  log('reviewers - new_reviewers', Array.from(new_reviewers).join(', '))
+
+  // Filter users to repository collaborators. If they're not, they can't be requested
+  // for review. In that case, they probably missed their invite to the maintainers team.
+  const reviewers = (
+    await Promise.all(
+      Array.from(new_reviewers, async (username) => {
+        try {
+          await github.rest.repos.checkCollaborator({
+            ...context.repo,
+            username,
+          })
+          return username
+        } catch (e) {
+          if (e.status !== 404) throw e
+          log(
+            `User ${username} cannot be requested for review because they don't exist or are not a repository collaborator, ignoring. Probably missed the automated invite to the maintainers team (see <https://github.com/NixOS/nixpkgs/issues/234293>).`,
+          )
+        }
+      }),
+    )
+  ).filter(Boolean)
+  log('reviewers - reviewers', reviewers.join(', '))
+
+  if (reviewers.length > 15) {
+    log(
+      `Too many reviewers (${reviewers.join(', ')}), skipping review requests.`,
+    )
+    // false indicates, that we do have reviewers and don't need the "needs: reviewers" label.
+    return false
+  }
```

Comment:
I know this is the old behavior, but maybe we should still request reviews in cases like this, maybe limiting to the "first" 15 reviewers (by some stable sorting method).